test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -520,7 +520,12 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			for _, currentPod := range pods.Pods {
 				log.Printf("Checking %s - ready: %t, restarts: %d", currentPod.Metadata.Name, currentPod.Status.ContainerStatuses[0].Ready, currentPod.Status.ContainerStatuses[0].RestartCount)
 				Expect(currentPod.Status.ContainerStatuses[0].Ready).To(BeTrue())
-				Expect(currentPod.Status.ContainerStatuses[0].RestartCount).To(BeNumerically("<", 5))
+				tooManyRestarts := 5
+				if strings.Contains(currentPod.Metadata.Name, "cluster-autoscaler") && !eng.HasWindowsAgents() {
+					log.Print("need to investigate cluster-autoscaler restarts on Linux!")
```

Comment:
nothing in kube-system runs on Windows, so I don't see any need for if eng.HasWindowsAgents(). I think whatever tests we run against deployments in kube-system should be the same regardless of which node OS are used.

TL;DR - I'm ok with 10 restarts everywhere

Reply:
+1